internal/controller/lifecycle/cluster_plan_utils.go

```diff
@@ -55,11 +57,25 @@ func mergeAnnotations(originalAnnotations map[string]string, targetAnnotations *
 	}
 }
 
-// mergeServiceAnnotations keeps the original annotations except prometheus scrape annotations.
+// mergeServiceAnnotations keeps the target annotations except prometheus scrape annotations.
```

Comment:
"mergeServiceAnnotations merge annotations from original to tartget, and also remove targetAnnotations' Prometheus scrape annotations if found."

Reply:
fixed